### ISSUE TITLE
List Of What Else Needs Done

### ISSUE THREAD

**Issue by Akiira** [Akiira/MUD-Server#1]
I am just writing a list so I can remember what else needs to be done, you don't need to respond to this Hand. But if you want to take one of these let me know here.
1. Allow clients to create a new character on the central server. These characters need to be rolled random stats and given the appropriate equipment for their race and class. (High Priority)
2. Add new rooms. We should have at least two unique areas so we can have a separate server running each area. This could be one server holding a town and another server holding a simple dungeon. (Med-High Priority)
3. Add new types of monsters (Med Priority)
4. Add loot and item drops to the monsters. (Priority)
5. Add some simple experience and level system. (lower priority)
6. Add class system to game (Lower priority)
7. Add magic system to game (Lower Priority)
8. Incorporate Christian's Algorithm into some system of the game. I think I will use it to do some bidding system. An item will be up for auction and near the end, to ensure we give it to the correct winner, use Christians algorithm to find the most appropriate winner. (High Priority)
9. We have redirect working from central server to world server, just need to make sure it also works between two world servers. This requires changing the Room class to have some special flag when a linked room is on a separate server. Then when a character tries to move to that room the client gets redirected. (Med-High Priority)
10. Possibly add some trading between players. Not sure if this is important to our grade or not. (Med-High Priority?)
11. Add missing commands to the game.

**Comment by Akiira** [comment]
Just updating the list of what needs done. Assuming all goes well when I test it tomorrow we are basically done with the auction. I don't think we should both storing a clients average ping because it works rather fast already, and we are not being graded on efficiency per se.

Again, if you see anything missing here feel free to add it, also if you want to take any of these just say so.
1. Allow clients to create a new character on the central server. These characters need to be rolled random stats and given the appropriate equipment for their race and class. (High Priority) (Low Difficulty)
2. Add new rooms. We should have at least two unique areas so we can have a separate server running each area. This could be one server holding a town and another server holding a simple dungeon. (Med-High Priority) (Med Difficulty)
3. Add new types of monsters (Med Priority) (Low Difficulty)
4. Add some simple experience and level system. (low priority) (Low-Med Difficulty)
5. Add class system to game (Low priority) (Med-High Difficulty)
6. Add magic system to game (Low Priority) (High Difficulty)
7. Possibly add some trading between players. Not sure if this is important to our grade or not. (Med-High Priority?) (Med-High Difficulty)

**Comment by mixisbad** [comment]
Respawn player  after dead ( Right now, player still can attack after dead)

**Comment by Akiira** [comment]
Hopefully this will be the last time I have to update this list. We are almost done. All the hard stuff is done.
1. Add new rooms. We should have at least two unique areas so we can have a separate server running each area. This could be one server holding a town and another server holding a simple dungeon. (Med-High Priority) (Med Difficulty)
2. Add new types of monsters (Med Priority) (Low Difficulty)
3. Add some simple experience and level system. (low priority) (Low-Med Difficulty)
4. Add class system to game (Low priority) (Med-High Difficulty)
5. Add magic system to game (Low Priority) (High Difficulty)
6. Perhaps do some formal testing on parts of the program
7. Perhaps write documentation in the code for everything, maybe in the format for godoc. (Low Priority) (Very Low Difficulty)

**Comment by mixisbad** [comment]
I am not quite understand how the item xml works for now. I cannot find the
quantity tag and somehow see multiple same item in the save file. Not sure
if it's the cause of the trade bug or not, will take a look at it further
more.

**Comment by Akiira** [comment]
Storing an item once, with a quantity tag would be possible, and better, but it makes un-marshaling slightly harder. Right now an item is stored once for each one a player has, so its not related to the trade bug. 

Also, I don't know if you saw, but I wrote an alternate trading implementation on the trade 2 branch that is working; maybe monday we can look at the different versions and decide which one we want to use. 

I also cleaned up and fixed a lot of stuff including some inventory functions; if you are still interested in working on your trade version it could possibly fix the bug to convert it over to the new inventory functions.

**Comment by mixisbad** [comment]
If it works then let's use your implementation. Btw, did you branch from my
implementation? I have asked some people to make sure the command is easy
enough to use. It doesn't matter what is the backend behind.

**Comment by Akiira** [comment]
I looked over your branch before starting mine but I did not branch off from it.

Since we are going to use mine then, I will make that branch the new master.

**Comment by Akiira** [comment]
fix auction when bid amounts are the same

**Comment by Akiira** [comment]
add name of current highest bid to auction message
